elrond-wasm/src/types/flags/code_metadata.rs

```diff
@@ -98,6 +107,37 @@ impl TypeAbi for CodeMetadata {
     }
 }
 
+impl SCDisplay for CodeMetadata {
+    fn fmt<F: FormatByteReceiver>(&self, f: &mut F) {
+        let mut metadata: bool = true;
+        if self.is_upgradeable() {
+            f.append_bytes(UPGRADEABLE_STRING);
+            metadata = false;
+        }
+        if self.is_readable() {
+            if !metadata {
+                f.append_bytes(b"|");
```

Comment:
`metadata = false;` also needed here.
It is also possible to extract this to a small private function above, to avoid code duplication.